lib/testing/matrix.js

```diff
@@ -0,0 +1,185 @@
+'use strict'; // eslint-disable-line strict
+
+/**
+* Running tests with multiple configurations using varying parameters
+* @constructor
+*/
+class TestMatrix {
+    /**
+    * Builds a new testing Matrix
+    *
+    * @param {Object.<string, string|string[]>} params - Argument of the matrix
+    *
+    * @returns {TestMatrix} new testing Matrix
+    */
+    constructor(params) {
+        /**
+        * Create deep copy for prevent the user to modify the object outside
+        * the class
+        **/
+        this.params = Object.assign({}, params);
+        this.paramsToString = JSON.stringify(this.params);
+        this.callback = null;
+        this.elementsToSpecialize = [];
+        this.listOfExceptions = [];
+    }
+
+    /**
+    * Callback used for a generated matrix
+    * Used by @generate and @if method
+    *
+    * @callback GenerateMatrixCallback
+    * @param {TestMatrix} matrix - The following generate matrix.
+    *
+    * @return {undefined}
+    */
+
+    /**
+    * Generate testing matrix, call execute method to run callback
+    *
+    * @param {string[]} elementsToSpecialize - key element we want to
+    *                    specialize on the matrix
+    * @param {GenerateMatrixCallback} callback - callback function
+    *                    who want to be call on execute method
+    *                    for the current generate matrix
+    *
+    * @returns {TestMatrix} Return current instance of Matrix object
+    */
+    generate(elementsToSpecialize, callback) {
+        this.elementsToSpecialize = elementsToSpecialize;
```

Comment:
Here you are assigning the param `elementsToSpecialize` which is a string to the array `this.elementsToSpecialize`. You should probably do a push into the array or initialize the property as a string, if you are planning on using only one element in the TestMatrix's instance lifetime.